### ISSUE TITLE
Cachix use "Empty stream" errors

### ISSUE THREAD

**Issue by thequicksort** [cachix/cachix#365]
Hi there, thank you for this project. 

My lab has a cachix account registered with [uwmisl.cachix.org
](uwmisl.cachix.org). 

However, upon trying to use it with 

```
cachix use uwmisl
```

We're getting the following error 

```
cachix: Data.Text.Internal.Fusion.Common.head: Empty stream
CallStack (from HasCallStack):
  error, called at libraries/text/src/Data/Text/Internal/Fusion/Common.hs:943:24 in text-1.2.4.1:Data.Text.Internal.Fusion.Common
```

I can't find any other debugging information unfortunately :( 

If it helps, here's a subsection of my printenv.  
```
PATH=/nix/var/nix/profiles/per-user/dna/profile/bin:/nix/var/nix/profiles/default/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Applications/VMware Fusion.app/Contents/Public:/Library/Apple/usr/bin:/Users/dna/.nix-profile/bin:/nix/var/nix/profiles/default/bin

TERM=xterm-256color
SHLVL=1
NIX_PROFILES=/nix/var/nix/profiles/default /Users/dna/.nix-profile
NIX_PATH=nixpkgs=/nix/var/nix/profiles/per-user/root/channels/nixpkgs:/nix/var/nix/profiles/per-user/root/channels
P9K_SSH=0
```

Any ideas about what's causing this? We've tried re-authenticating with the auth token, and the result is the same

**Comment by domenkozar** [comment]
Hi @thequicksort,

Could you paste the output of `id -Gn $USER` in your terminal?

**Comment by thequicksort** [comment]
Sure thing, thanks again for your help! Running `id -Gn $USER` on my MacBook results in:

```
staff everyone localaccounts _appserverusr admin _appserveradm _lpadmin com.apple.access_ssh com.apple.sharepoint.group.1 com.apple.sharepoint.group.2 _appstore _lpoperator _developer _analyticsusers com.apple.access_ftp com.apple.access_screensharing

```

I can try from my Linux machine as well (though they're all getting the same error)

**Comment by domenkozar** [comment]
@thequicksort that looks good, how about `nix show-config | grep trusted-users`?

**Comment by thequicksort** [comment]
> @thequicksort that looks good, how about `nix show-config | grep trusted-users`?

Sorry for my delayed reply. 
```
nix show-config | grep trusted-users
trusted-users = root
```

I added my `$(whoami)` to my trusted-users but I'm running into the same issue.

**Comment by domenkozar** [comment]
@thequicksort could you try installing cachix with:

```
nix-env -if https://github.com/cachix/cachix/tarball/fix-head-error --substituters 'https://cache.nixos.org https://cachix.cachix.org' --trusted-public-keys 'cachix.cachix.org-1:eWNHQldwUO7G2VkjpnjDbWwy4KQ/HNxht7H4SSoMckM= cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY='
```

That might fix your problem, PR at https://github.com/cachix/cachix/pull/366

**Comment by thequicksort** [comment]
Thanks again! I ended up deleting my /etc/nix.conf and ~/.config/nix/nix.conf files, then re-installed Nix, then reinstalled cachix with sudo (and also used sudo for configuring the caches). 

For future people running into this: This may be due to a misconfigured Nix Conf.

I could be wrong here, as I'm still learning about the single-user vs. multi-user nix installs and how the configurations interact, but what I suspect I did wrong originally was : 

1) Installed Cachix as a non-root user, so it wasn't available outside. I changed this [here](https://github.com/uwmisl/poretitioner/blob/jdunstan/beta/bootstrap_dev.sh#L312). 

2) Called `cachix use cachix` as a non-root user in a multi-user install. 

I imagine this led to cachix not being whitelisted for my user, which may have led to a network issue. If that's the case, would the PR you mentioned fix it? I don't know enough Haskell to understand the change, unfortunately.

**Comment by domenkozar** [comment]
@thequicksort did my fix help at all or you didn't try it before doing the reinstallation?

**Comment by domenkozar** [comment]
Note there's a typo in https://github.com/uwmisl/poretitioner/blob/jdunstan/beta/bootstrap_dev.sh#L312

**Comment by domenkozar** [comment]
@thequicksort if you didn't try out my fix that's OK, I've merged the fix that would solve the problem in case nix.conf wasn't parseable.

**Comment by domenkozar** [comment]
This is fixed on master and will be part of Cachix 0.6.1, let me know if the issue persists.

**Comment by domenkozar** [comment]
I've released Cachix 0.6.1 that fixes this!
